var/spack/repos/builtin/packages/bat/package.py

```diff
@@ -4,26 +4,38 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
+import sys
 
 
 class Bat(Package):
     """A cat(1) clone with wings."""
 
     homepage = "https://github.com/sharkdp/bat"
-    url      = "https://github.com/sharkdp/bat/archive/v0.10.0.tar.gz"
+    url      = "https://github.com/sharkdp/bat/archive/v0.13.0.tar.gz"
 
+    version('0.13.0', sha256='f4aee370013e2a3bc84c405738ed0ab6e334d3a9f22c18031a7ea008cd5abd2a')
     version('0.12.1', sha256='1dd184ddc9e5228ba94d19afc0b8b440bfc1819fef8133fe331e2c0ec9e3f8e2')
 
     depends_on('rust')
 
+    def url_for_version(self, version):
+        url = "https://github.com/sharkdp/bat/archive/v{0}.tar.gz"
+        return url.format(version)
```

Comment:
This shouldn't be necessary

Reply:
Why is the URL not necessary, for example if I want to install version 0.11 or something, can spack handle the URL automatically?

Reply:
The `url` attribute is mandatory to download something, but Spack is clever enough to determine which url each version needs:
```console
$ spack url parse https://github.com/sharkdp/bat/archive/v0.12.1.tar.gz
==> Parsing URL: https://github.com/sharkdp/bat/archive/v0.12.1.tar.gz

==> Matched version regex  1: r'^v?(\\d[\\d._-]*)$'
==> Matched  name   regex  0: r'github\\.com/[^/]+/([^/]+)'

==> Detected:
    https://github.com/sharkdp/bat/archive/v0.12.1.tar.gz
                               ---          ~~~~~~
    name:    bat
    version: 0.12.1

==> Substituting version 9.9.9b:
    https://github.com/sharkdp/bat/archive/v9.9.9b.tar.gz
                               ---          ~~~~~~
```
Overriding `url_for_version` is only necessary when the URL is weird.

Reply:
Oh I understand, I remove this change.